core/lib/runner.js

```diff
@@ -405,6 +405,10 @@ function runScenario(script, intermediate, runState, contextVars) {
   const scenarioStartedAt = process.hrtime();
   const scenarioContext = createContext(script, contextVars);
 
+  if(script.scenarios[i].apiVersions){
+    scenarioContext.vars.apiVersions = script.scenarios[i].apiVersions;
```

Comment:
i would put this logic in the `createContext` function since it's modifying the context vars which is what that function does